sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -182,309 +183,79 @@ private static String namespaceOf(ClassForDisplay clazz) {
-     *
-     * @see DisplayData.Builder#add(String, Class)
-     */
-    ItemBuilder addIfNotNull(String key, @Nullable Class<?> value);
-
-    /**
-     * Register the given class display data if the value is not null.
-     *
-     * @see DisplayData.Builder#add(String, ClassForDisplay)
+     * Register the given display item if the value is not null.
      */
-    ItemBuilder addIfNotNull(String key, @Nullable ClassForDisplay value);
+    Builder addIfNotNull(Item<?> item);
 
     /**
-     * Register the given class display data if the value is different than the specified default.
-     *
-     * @see DisplayData.Builder#add(String, Class)
-     */
-    ItemBuilder addIfNotDefault(
-        String key, @Nullable Class<?> value, @Nullable Class<?> defaultValue);
-
-    /**
-     * Register the given class display data if the value is different than the specified default.
-     *
-     * @see DisplayData.Builder#add(String, ClassForDisplay)
+     * Register the given display item if the value is different than the specified default.
      */
-    ItemBuilder addIfNotDefault(
-        String key, @Nullable ClassForDisplay value, @Nullable ClassForDisplay defaultValue);
-  /**
-   * Register the given display data with the specified type.
-   *
-   * <p> The added display data is identified by the specified key and namespace from the current
-   * transform or component.
-   *
-   * @throws ClassCastException if the value cannot be safely cast to the specified type.
-   *
-   * @see DisplayData#inferType(Object)
-   */
-    ItemBuilder add(String key, Type type, Object value);
+    <T> Builder addIfNotDefault(Item<T> item, @Nullable T defaultValue);
   }
 
   /**
-   * Utility to append optional fields to display data, or register additional display data
-   * items.
+   * {@link Item Items} are the unit of display data. Each item is identified by a given key
+   * and namespace from the component it belongs to.
+   *
+   * <p>{@link Item Items} are registered via {@link DisplayData.Builder#add}
+   * within {@link HasDisplayData#populateDisplayData} implementations.
    */
-  public interface ItemBuilder extends Builder {
-    /**
-     * Add a human-readable label to describe the most-recently added metadata field.
-     * A label is optional; if unspecified, UIs should display the metadata key to identify the
-     * display item.
-     *
-     * <p>Specifying a null value will clear the label if it was previously defined.
-     */
-    ItemBuilder withLabel(@Nullable String label);
+  @AutoValue
+  public abstract static class Item<T> {
 
     /**
-     * Add a link URL to the most-recently added display data. A link URL is optional and
-     * can be provided to point the reader to additional details about the metadata.
-     *
-     * <p>Specifying a null value will clear the URL if it was previously defined.
+     * The namespace for the display item. The namespace is generated from the component which
```

Comment:
We can stick with this now. Once we clarify the relationship between `super.populateDisplayData(builder)` and `builder.include(super)`, it may be easier to come up with a clear description.